web/src/games/bombsAndBunnies/components/PlayerZones.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+
+import css from './PlayerZones.css';
+import { IPlayerZoneProps, PlayerZone, PlayerStatus } from './PlayerZone';
+import IPlayer from '../player';
+import { getMaxPlayerBet } from '../game';
+import { BetDisplay, IBetDisplayProps } from './BetDisplay';
+import { DiscardPile, IDiscardPileProps } from './DiscardPile';
+
+export interface IPlayerZonesProps {
+  betDisplayProps?: IBetDisplayProps;
+  discardPileProps?: IDiscardPileProps;
+
+  currentPlayerId: string;
+  perspectivePlayerId: string;
+  players: IPlayer[];
+  canRevealTargetStack: (playerId: string) => boolean;
+  revealCard?: (playerId: string) => void;
+}
+
+export class PlayerZones extends React.Component<IPlayerZonesProps, {}> {
+  render() {
+    return (
+      <div className={css.playerZonesContainer}>
+        <div className={css.playerZones}>{this.renderZones()}</div>
+        <div className={css.centerDisplay}>
+          {this.renderBetDisplay(0)}
+          {this.renderDiscardPile(1)}
+        </div>
+      </div>
+    );
+  }
+
+  renderBetDisplay(key: number) {
+    var props = this.props.betDisplayProps;
+    if (props === undefined) return null;
+
+    return <BetDisplay key={key} {...this.props.betDisplayProps}></BetDisplay>;
+  }
+
+  renderDiscardPile(key: number) {
+    var props = this.props.discardPileProps;
+    if (this.props.betDisplayProps !== undefined || props === undefined) return null;
+
+    return <DiscardPile key={key} {...this.props.discardPileProps}></DiscardPile>;
+  }
+
+  renderZones() {
+    var players = this.props.players;
+    var perspectiveIndex = players.findIndex((p) => p.id === this.props.perspectivePlayerId);
+    var totalPlayers = this.props.players.length;
+    var zones = this.props.players.map((p: IPlayer, i) => {
+      var result: IPlayerZoneProps = {
```

Comment:
A suggestion to make this code a tiny bit more readable, especially for the attributes that have long computation.... You could move them to a `const` definition before this line in their own line, and then later just do `const result: IPlayerZoneProps = { playerStatuses, bet, totalPlayerCards }` ...

Reply:
I can have a look at making it more readable, but I don't follow about making them const? Could you provide an example?

Reply:
This was more nitpicking than anything, but my idea was something like this:
```
const totalPlayerCards = p.hand.length + p.stack.length + p.revealedStack.length;
const bet = p.bet;
const result: IPlayerZoneProps = { bet, totalPlayerCards }
```
You can skip this if you want

Reply:
Thanks @flamecoals, I've made those changes. It will take a bit of getting used to, but I can see why you'd do it